### PR TITLE
🎏 Limit pagination on Contact.list to 10000 contacts

### PR DESCRIPTION
**Pull request by dodobas**
Fixes: https://github.com/rapidpro/rapidpro/issues/876

### CODE REVIEW

File: temba/utils/views.py [hunk 1]
```diff
@@ -151,6 +153,18 @@ class ContactListPaginator(Paginator):
...
 
+    ES_SEARCH_BUFFER_SIZE = 10000
+
+    @cached_property
+    def num_pages(self):
+
+        if isinstance(self.object_list, ModelESSearch):
+            # limit maximum number of pages when searching contacts on ES
+            # https://github.com/rapidpro/rapidpro/issues/876
+            return ceil(self.ES_SEARCH_BUFFER_SIZE / self.per_page)
```
Comment (rowanseymour) [inline temba/utils/views.py]: better to use `int(..)` so method returns an `int`

Comment (dodobas) [reply temba/utils/views.py]: `ceil` also returns an int, can you explain the intention of your comment

Comment (rowanseymour) [reply temba/utils/views.py]: oh weird looks like they changed that in Python 3. `ceil` used to return a `float`

Comment (rowanseymour) [reply temba/utils/views.py]: tho don't you still want `int()` here? Suppose `ES_SEARCH_BUFFER_SIZE=100` and `self.per_page=30` then this evaluates to `4` pages of `30` so you'll fetch up to `120` contacts.. and won't that blow up?

Comment (dodobas) [reply temba/utils/views.py]: I would say that it's going to blow up, we can always catch the exception and return an empty resultset, or whatever was in the resultset up to that moment (if it's possible)

The code is loosely based on the original Paginator code (https://github.com/django/django/blob/master/django/core/paginator.py#L100) ... and it uses `ceil` ... `int` behaves basically the same as `floor` for positive numbers ...

Just for reference:
Page 1 ... 0-30
Page 2 ... 30-60
Page 3 ... 60-90
Page 4 ... 90-120

`4` is still correct, we start counting pages from 1, and not 0 ... although ES `,search()` iterator would likely blow up when it get's to the end

Comment (rowanseymour) [reply temba/utils/views.py]: But in the Django case it's going to return a final page with that has less than `page_size` items. Do we have any code to check that our final page doesn't try to look beyond ES_SEARCH_BUFFER_SIZE ?

Comment (dodobas) [reply temba/utils/views.py]: we do now :)

added new feature: ACTUALLY use the actual number of results to actually correctly calculate the max_page number based on actual results, actually ... for any ES search it was defaulting to ES_SEARCH_BUFFER_SIZE ... https://github.com/nyaruka/rapidpro/pull/2368/files#diff-6e00c2a36e167deba94083a682063c3dR165

**Pull request merged**